backend/core/recipes/views.py

```diff
@@ -142,48 +154,50 @@ def copy(self, request: Request, pk: str) -> Response:
         )
 
 
-class TeamRecipesViewSet(
-    viewsets.GenericViewSet,
-    mixins.RetrieveModelMixin,
-    mixins.DestroyModelMixin,
-    mixins.UpdateModelMixin,
-):
+def parse_int(val: str) -> Optional[int]:
+    try:
+        return int(val)
+    except ValueError:
+        return None
 
-    serializer_class = RecipeSerializer
+
+@api_view(["GET"])
+@permission_classes((IsAuthenticated,))
+def get_recipe_timeline(request: Request, recipe_pk: int) -> Response:
+    user: MyUser = request.user
+    team = user.selected_team
+
+    recipe = get_object_or_404(Recipe, pk=recipe_pk)
+
+    if not has_recipe_access(recipe=recipe, user=user):
```

Comment:
Why not move this logic into a query set manager so that only recipes visible to a user are returned in the view?